tests/integration/s3/test_s3.py

```diff
@@ -4752,6 +4752,38 @@ def test_s3_intelligent_tier_config(self, aws_client, s3_create_bucket, snapshot
         response = aws_client.s3.list_bucket_intelligent_tiering_configurations(Bucket=bucket)
         snapshot.match("list_bucket_intelligent_tiering_configurations_3", response)
 
+    @pytest.mark.aws_validated
+    def test_s3_get_object_headers(self, aws_client, s3_create_bucket, snapshot):
+        bucket = s3_create_bucket()
+        key = "en-gb.wav"
+        file_path = os.path.join(f"../files/{key}")
+        aws_client.s3.upload_file(file_path, bucket, key)
+        objects = aws_client.s3.list_objects(Bucket=bucket)
+        etag = objects["Contents"][0]["ETag"]
+
+        try:
+            aws_client.s3.get_object(Bucket=bucket, Key=key, IfNoneMatch=etag)
+        except ClientError as e:
+            snapshot.match("if_none_match_err_1", e.response)
+        try:
+            aws_client.s3.get_object(Bucket=bucket, Key=key, IfNoneMatch=etag.strip('"'))
+        except ClientError as e:
+            snapshot.match("if_none_match_err_2", e.response)
```

Comment:
nit: what I was interested with those tests would be to see if AWS would return additional headers in the 304 error.
After a quick test, it seems it only returns the `etag` and `last-modified` headers from the object, nothing else except `x-amz-id-2` and `x-amz-request-id` which we already return. 
Returned value: 
```python
{'Error': {'Code': '304', 'Message': 'Not Modified'}, 'ResponseMetadata': {'RequestId': 'M1HKSF3MBSFDGW5X', 'HostId': 'Y30wm4vnx1fsQLgCjpW2xu0vsAzwal5ITs2GOnqRbmTmcekr2PTtgaSNlKTleiCZRzUY7T6OkMQ=', 'HTTPStatusCode': 304, 'HTTPHeaders': {'x-amz-id-2': 'Y30wm4vnx1fsQLgCjpW2xu0vsAzwal5ITs2GOnqRbmTmcekr2PTtgaSNlKTleiCZRzUY7T6OkMQ=', 'x-amz-request-id': 'M1HKSF3MBSFDGW5X', 'date': 'Wed, 05 Jul 2023 19:19:51 GMT', 'last-modified': 'Wed, 05 Jul 2023 19:19:48 GMT', 'etag': '"6ac5910e18a3566fe197c4380baaf7a0"', 'server': 'AmazonS3'}, 'RetryAttempts': 0}}
```
Maybe we could add a TODO or something for us to return those values at some point in the 304 exception.